scene/resources/mesh.h

```diff
@@ -159,11 +159,43 @@ class Mesh : public Resource {
 	Size2i get_lightmap_size_hint() const;
 	void clear_cache() const;
 
-	typedef Vector<Vector<Face3>> (*ConvexDecompositionFunc)(const Vector<Face3> &p_faces, int p_max_convex_hulls);
+	struct ConvexDecompositionSettings {
+		enum Mode : int {
+			CONVEX_DECOMPOSITION_MODE_VOXEL = 0,
+			CONVEX_DECOMPOSITION_MODE_TETRAHEDRON
+		};
+
+		/// Maximum concavity. [Range: 0.0 -> 1.0]
+		real_t max_concavity = 1.0;
+		/// Controls the bias toward clipping along symmetry planes. [Range: 0.0 -> 1.0]
+		real_t symmetry_planes_clipping_bias = 0.05;
+		/// Controls the bias toward clipping along revolution axes. [Range: 0.0 -> 1.0]
+		real_t revolution_axes_clipping_bias = 0.05;
+		real_t min_volume_per_ch = 0.0001;
+		/// Maximum number of voxels generated during the voxelization stage.
+		uint32_t resolution = 10'000;
+		uint32_t max_num_vertices_per_ch = 32;
+		/// Controls the granularity of the search for the "best" clipping plane.
+		/// [Range: 1 -> 16]
+		uint32_t plane_downsampling = 4;
+		/// Controls the precision of the convex-hull generation process during the
+		/// clipping plane selection stage.
+		/// [Range: 1 -> 16]
+		uint32_t convexhull_downsampling = 4;
+		/// enable/disable normalizing the mesh before applying the convex decomposition.
+		bool normalize_mesh = false;
+		Mode mode = CONVEX_DECOMPOSITION_MODE_VOXEL;
+		bool convexhull_approximation = true;
+		bool ocl_acceleration = true;
+		/// This is the maximum number of convex hulls to produce from the merge operation.
+		uint32_t max_convex_hulls = 1;
```

Comment:
The `Simple Convex` approach always generates complex shapes than VHACD is able to generate. I setup VHACS so that it generates the most simple possible collision shape for the given Mesh, so that you can increase the precision later. I would like to keep it as is, so it would be much easier to use for most of the cases, if this approach sounds good to you.

Reply:
I see where you're coming from, and it makes sense with some use cases, but users will think the physics shape importer is broken if it says convex decomposition by default, but actually generates one shape that doesn't seem to follow the geometry.

Isn't decomposition the most common case that you want? If I import the physics for environment or objects, I would expect to get a collision that matches the mesh, either using a triangle mesh shape (more accurate but limited to static) or convex decomposition (balanced compromise that works for dynamic objects too).